### PR TITLE
[RFR] Refactor View.fields

### PR DESCRIPTION
**Pull request by ThieryMichel**
- [x] Make view.fields return an array instead of a literal
- [x] Prepare to remove Field.order method by using it only in View.fields
- [x] add deprecation  warning

### CODE REVIEW

**Comment by jpetitcolas** [comment]
Can you also update changelog and README to show depreciations?

File: src/javascripts/ng-admin/es6/lib/Entry.js [hunk 1]
```diff
@@ -22,13 +22,14 @@ class Entry {
         var identifier = view.identifier();
         var identifierValue = null;
 
-        var values = restEntry;
-        for (var fieldName in view.fields()) {
-            var field = view.fields()[fieldName];
-            if (field.name() in restEntry) {
-                values[fieldName] = field.getMappedValue(restEntry[field.name()], restEntry);
+        // var values = restEntry;
```
Comment (fzaninotto) [inline src/javascripts/ng-admin/es6/lib/Entry.js]: Debug spotted

File: src/javascripts/ng-admin/es6/lib/View/View.js [hunk 2]
```diff
@@ -89,17 +89,20 @@ class View {
      * fields({Field2, Field3})
      */
     fields() {
-        if (!arguments.length) return View._indexFieldsByName(this._fields);
+        if (!arguments.length) {return this._fields;}
```
Comment (fzaninotto) [inline src/javascripts/ng-admin/es6/lib/View/View.js]: Loose the curly braces here.

File: src/javascripts/ng-admin/es6/lib/View/View.js [hunk 3]
```diff
@@ -89,17 +89,20 @@ class View {
...
 
         [].slice.call(arguments).map(function(argument) {
             View.flatten(argument).map(arg => this.addField(arg));
-        }, this)
+        }, this);
+
+        this._fields = this._fields.sort((a, b) => (a.order() - b.order()));
```
Comment (fzaninotto) [inline src/javascripts/ng-admin/es6/lib/View/View.js]: I think this line should be moved to `addField()` instead.

File: src/javascripts/ng-admin/es6/lib/View/View.js [hunk 4]
```diff
@@ -195,10 +191,10 @@ class View {
...
     processFieldsDefaultValue(entry) {
-        for (var i in this._fields) {
-            var field = this._fields[i];
+
+        this._fields.forEach(function (field) {
             entry.values[field.name()] = field.defaultValue();
-        }
+        });
 
         return this;
```
Comment (fzaninotto) [inline src/javascripts/ng-admin/es6/lib/View/View.js]: Not your fault, but this method should return the modified `entry`, not the current view.

File: src/javascripts/ng-admin/es6/lib/View/View.js [hunk 5]
```diff
@@ -221,25 +214,10 @@ class View {
...
-            var field = fields[i];
-            result[field.name()] = field;
-        }
-
-        return result;
+        return this._fields.filter(f => f.type() === type);
     }
 
     addField(field) {
-        if (field.order() === null) {
```
Comment (fzaninotto) [inline src/javascripts/ng-admin/es6/lib/View/View.js]: I think you still need this.

**Comment by fzaninotto** [comment]
Great job, this really simplifies fields handling, it's much more intuitive.

**Comment by fzaninotto** [comment]
Thanks!

**Comment by fzaninotto** [comment]
How about `fieldsUtils`? Is it still required?

**Pull request merged**